src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -121,34 +116,34 @@ OpResult CondTraverseConsume(OpBase *opBase, Record *r) {
 
         OpResult res = child->consume(child, r);
         if(res != OP_OK) return res;
-        _extractColumn(op, *r);
+        _extractColumn(op, r);
     }
 
     /* Get node from current column. */
-    Graph_GetNode(op->graph, dest_id, op->algebraic_expression->dest_node);
+    Node *destNode = Record_GetNode(r, op->destNodeRecIdx);
+    Graph_GetNode(op->graph, dest_id, destNode);
 
     if(op->algebraic_expression->edge != NULL) {
         // We're guarantee to have at least one edge.
         Node *srcNode;
         Node *destNode;
-
         size_t operandCount = op->algebraic_expression->operand_count - 1;
+
         if(op->algebraic_expression->operands[operandCount].transpose) {
-            srcNode = op->algebraic_expression->dest_node;
-            destNode = op->algebraic_expression->src_node;
+            srcNode = Record_GetNode(r, op->destNodeRecIdx);
+            destNode = Record_GetNode(r, op->srcNodeRecIdx);
         } else {
-            srcNode = op->algebraic_expression->src_node;
-            destNode = op->algebraic_expression->dest_node;
+            srcNode = Record_GetNode(r, op->srcNodeRecIdx);
+            destNode = Record_GetNode(r, op->destNodeRecIdx);
         }
 
         Graph_GetEdgesConnectingNodes(op->graph,
                                       ENTITY_GET_ID(srcNode),
                                       ENTITY_GET_ID(destNode),
                                       op->edgeRelationType,
                                       &op->edges);
-        return _CondTraverse_SetEdge(op, r);
```

Comment:
What if `_CondTraverse_SetEdge` returns OP_DEPLETED?

Reply:
Impossible, if we're here then we've found a dest node from src node, and that's only possible if there exists an edge connecting the two, see comment at line 127,
additional calls to consume will be fulfilled by line 104 until all edges connecting src to dest are consumed, in which case a new set of edges is computed if such exists

Reply:
:+1: